eudemo/eudemo/maintenance/equatorial_port.py

```diff
@@ -0,0 +1,229 @@
+        """
+        x_ib = self.x_ib
+        x_ob = self.x_ob
+        z_h = self.params.ep_height.value + (2 * self.koz_offset)
+
+        x = (x_ib, x_ob, x_ob, x_ib)
+        z = (-z_h / 2, -z_h / 2, z_h / 2, z_h / 2)
+
+        ep_boundary = BluemiraFace(
+            make_polygon({"x": x, "y": 0, "z": z}, label="koz", closed=True)
+        )
+        return ep_boundary
+
+
+class CastellationBuilder(Builder):
+    """
+    Castellation Builder
+    """
+
+    NAME = "Castellation"
+    param_cls: Type[CastellationBuilderParams] = CastellationBuilderParams
+
+    def __init__(
+        self,
+        params: Union[Dict, ParameterFrame, CastellationBuilderParams],
+        build_config: Union[Dict, None],
+        length: float,
+        start_profile: BluemiraFace,
+        extrude_vec: tuple,
+        offsets: list,
+        castellation_offsets: list,
+    ):
+        super().__init__(params, build_config)
+        self.length = length
+        self.face = start_profile
+        self.vec = np.array(extrude_vec)
+        self.off = offsets
+        self.cst = castellation_offsets
+
+    def build(self) -> Component:
+        """
+        Build the castellated component.
+        """
+        # TODO: Implement corner radii using PR #1992
+        self.r_rad = self.params.cst_r_corner.value
+        self.n = self.params.n_components.value
+
+        xyz_solid = self.build_castellations(
+            self.face, self.vec, self.length, self.off, self.cst
+        )
+        xz_plane = BluemiraPlane(axis=(0, 1, 0))
+        xy_plane = BluemiraPlane(axis=(0, 0, 1))
+        xz_slice = BluemiraFace(slice_shape(xyz_solid, xz_plane))
+        xy_slice = BluemiraFace(slice_shape(xyz_solid, xy_plane))
+
+        return self.component_tree(
+            xz=[self.build_xz(xz_slice)],
+            xy=self.build_xy(xy_slice),
+            xyz=self.build_xyz(xyz_solid),
+        )
+
+    def build_xz(self, xz: BluemiraFace) -> PhysicalComponent:
+        """
+        Build the xz representation of the castellated component
+        """
+        body = PhysicalComponent(self.NAME, xz)
+        body.plot_options.face_options["color"] = BLUE_PALETTE["VV"][0]
+        return body
+
+    def build_xy(self, xy: BluemiraFace, n: int = 10) -> List[PhysicalComponent]:
+        """
+        Build the xy representation of the castellated component
+        """
+        body = PhysicalComponent(self.NAME, xy)
+        body.plot_options.face_options["color"] = BLUE_PALETTE["VV"][0]
+        return circular_pattern_component(body, n_children=n)
+
+    def build_xyz(self, xyz: BluemiraSolid, n: int = 10) -> List[PhysicalComponent]:
+        """
+        Build the xyz representation of the castellated component
+        """
+        cst_shape = PhysicalComponent(self.NAME, xyz)
+        cst_shape.display_cad_options.color = BLUE_PALETTE["VV"][0]
+        return circular_pattern_component(cst_shape, n_children=n)
```

Comment:
@oliverfunk is this still how we do sectored stuff nowadays?

Reply:
@IChiang99 Bluemira now works by patterning xyz components during show_cad because it cuts down on runtime and let's you specify the number of sectors you wish to view (see: https://github.com/Fusion-Power-Plant-Framework/bluemira/blob/ab336083030976071a4fab9f0b8c26a92acabdab/bluemira/base/reactor.py#L147)

build_xyz should return just 1 castellation, positioned correctly within a single sector.
The sectors are defined by n_TF, sector angle by 360/n_TF.